src/graphnet/models/detector/detector.py

```diff
@@ -97,6 +97,15 @@ def nb_outputs(self) -> int:
     def _validate_features(self, data: Data):
         if isinstance(data, Batch):
             data_features = [features[0] for features in data.features]
+            try:
+                assert data_features == [
+                    self.features[0] for i in range(len(data_features))
+                ]  # if true, data_features is a repeat of self.features[0] len(data) times - unwanted
+                data_features = data.features[
+                    0
+                ]  # if assert is true, data_features are set correctly
+            except AssertionError:
+                pass  # if the assert fails, data_features are not transposed.
```

Comment:
A rewrite without a try-except block makes the intention clearer for me, at least. That is: If the number of feature names in `data.features[0]` is the same as the batch size (i.e., `data.num_graphs`) and if all of those feature names are the same (`len(set(...)) = 1`) then the features are "transposed"; if not, we should be able to get them directly.
```suggestion
            # `data.features` is "transposed" and each list element contains
            # only duplicate entries.
            if (
                len(data.features[0]) == data.num_graphs
                and len(set(data.features[0])) == 1
            ):
                data_features = [features[0] for features in data.features]
            
            # `data.features` is not "transposed" and each list element
            # contains the original features.
            else:
                data_features = data.features[0]
```